lib/craft/clock_bound/client.ex

```diff
@@ -0,0 +1,233 @@
+    ```elixir
+    case ClockBound.Client.now() do
+      {:ok, clock_data} ->
+        IO.inspect(clock_data)
+
+      {:error, reason} ->
+        _
+    end
+    ```
+  2. Call `ClockBound.Client.compare/2` to compare two ClockBound windows:
+    ```elixir
+    case ClockBound.Client.compare(clock_data_1, clock_data_2) do
+      :lt -> IO.puts("Clock 1 is less than Clock 2")
+      :gt -> IO.puts("Clock 1 is greater than Clock 2")
+      :ov -> IO.puts("Clock 1 and Clock 2 overlap")
+    end
+    ```
+
+  ## Notes
+
+  - The default shared memory path is `/var/run/clockbound/shm`. You can override this by passing a custom path to the functions.
+  - The module adheres to the ClockBound protocol specification, which can be found [here](https://github.com/aws/clock-bound/blob/main/docs/PROTOCOL.md).
+  """
+
+  @default_shm_path "/var/run/clockbound/shm"
+  @clockbound_restart_grace_period 5 * 1_000_000_000
+
+  # 'AMZN CB'
+  @magic_0 0x414D5A4E
+  @magic_1 0x43420200
+
+  defstruct [
+    :earliest,
+    :latest,
+    :clock_status
+  ]
+
+  @type clock_status :: :unknown | :synchronized | :free_running | :invalid
+  @type t :: %__MODULE__{
+          earliest: DateTime.t(),
+          latest: DateTime.t(),
+          clock_status: clock_status()
+        }
+
+  @doc """
+  Returns the current clock time with error bounds and clock status.
+  """
+  def now(shm_path \\ @default_shm_path) do
+    with {:ok, data} <- read(shm_path),
+         real = :os.system_time(:nanosecond),
+         monotonic = monotonic_time(),
+         {:ok, {earliest, latest, clock_status}} <- compute_bound_at(real, monotonic, data),
+         {:ok, earliest} <- DateTime.from_unix(earliest, :nanosecond),
+         {:ok, latest} <- DateTime.from_unix(latest, :nanosecond) do
+      {:ok,
+       %__MODULE__{
+         earliest: earliest,
+         latest: latest,
+         clock_status: clock_status
+       }}
+    else
+      {:error, reason} -> {:error, reason}
+      _ -> {:error, :unknown}
+    end
+  end
+
+  @doc """
+  Compares two ClockBound windows and returns:
+
+    * `:lt` if a is strictly younger than b
+    * `:gt` if a is strictly older than b
+    * `:ov` if the ranges overlap and no strict ordering can be determined
+  """
+  def compare(cbe_1, cbe_2) do
+    cond do
+      DateTime.after?(cbe_1.earliest, cbe_2.latest) -> :gt
+      DateTime.before?(cbe_1.latest, cbe_2.earliest) -> :lt
+      true -> :ov
+    end
+  end
+
+  @doc """
+  Returns the current monotonic time in nanoseconds
+  For macOS, it uses the native NIF to get the monotonic time.
+  For other platforms, it uses the Erlang system_info function.
+  """
+  def monotonic_time do
+    case :os.type() do
+      {:unix, :darwin} ->
+        to_nanosecond(ClockBound.Native.get_monotonic_time())
+
+      _ ->
+        :erlang.system_info(:os_monotonic_time_source)[:time]
+    end
+  end
```

Comment:
instead of doing os detection, we should do capability detection at startup. i.e. checking if the `:clock_id` is `CLOCK_MONOTONIC`.